aospy/calc.py

```diff
@@ -587,6 +587,9 @@ def compute(self, write_to_tar=True):
         reduced = self._apply_all_time_reductions(full, monthly, eddy)
         logging.info("Writing desired gridded outputs to disk.")
         for dtype_time, data in reduced.items():
+            _add_units_and_description_attrs(data, self.var.units,
```

Comment:
Maybe it's just me, but for some reason the fact that this function doesn't return anything (and just mutates the input xarray object) feels odd.  It would feel more readable to me if this line were something along the lines of:
```python
data = _add_units_and_description_attrs(data, self.var.units, self.var.description, self.dtype_out_vert)
```
So I might consider re-writing the functions such that this would be possible (i.e. add `return` statements where appropriate).